### ISSUE TITLE
Markers not detected

### ISSUE THREAD

**Issue by mehditlili** [pal-robotics/aruco_ros#20]
I switched from ar track alvar to aruco due to some issues with camera model and the matrix used to compute the marker pose but I am now having problems getting aruco to detect the markers. Here is one example of the thresholded image using default parameters:
![aruco](https://cloud.githubusercontent.com/assets/7859307/18086389/e5f30ffc-6eaf-11e6-9cdd-295233cb5ce5.png)
Why is this marker not being detected??

**Comment by bmagyar** [comment]
I think the one in the middle should work if the frame was a rectangle. On this image it looks more like a puzzle piece. Could you also attach the original image please?

As for the markers on the side, I'm not sure if this is not too much distortion...

Have you tried tweaking the parameters for your system using rqt_reconfigure?

**Comment by jordi-pages** [comment]
Have you modified file single.launch and included the right marker ID and its proper size?

**Comment by mehditlili** [comment]
Yes the marker id and size were set correctly, the markers on the bottom are irrelevant here (alvar markers). The frame shape around the marker should be irrelevant too?, it is just white paper fixed with white double sided tape on a metal piece.
![aruco2](https://cloud.githubusercontent.com/assets/7859307/18086801/ff06f43e-6eb1-11e6-9c84-0f34dc750619.png)

**Comment by mehditlili** [comment]
Here is two comparisons, there is a difference of 3 cm between both positions.
![detected](https://cloud.githubusercontent.com/assets/7859307/18088039/7bbaca90-6eb8-11e6-908f-463708e5705d.png)
![not_detected](https://cloud.githubusercontent.com/assets/7859307/18088041/81ae63bc-6eb8-11e6-9a1b-4490a0538fe3.png)

**Comment by jordi-pages** [comment]
I would first try to detect another marker, just to make sure that AruCo does not have any problem with this specific marker.

**Comment by bmagyar** [comment]
Good idea. Give a shot to the same scene with this marker (this is the most balanced one):
https://github.com/pal-robotics/aruco_ros/blob/master/aruco_ros/etc/marker582_5cm.jpg

As a sidenote: this coming weekend I intend to make a ROS Kinetic package of `aruco` with the new version of the library, that may solve your issue.

**Comment by lettersfromfelix** [comment]
Is there a solution yet for the problem that markers are not recognized from a further than a specific distance? I ran a simulation, and when the camera is further away than 1m from a 6.5cm tag, it hasn't got any chance to detect it. Other libraries like apriltag has no problem at all with more than double the distance!
